### DISCUSSION TITLE
What is Standard Basic Frontmatter to support (or document)

### DISCUSSION THREAD

**Discussion by rufuspollock** [flowershow/flowershow#326] [General and Help]
When creating markdown pages I want to know about some standard frontmatter fields for common functionality e.g. title, date etc so that I can add them and get related functionality

Questions

* [ ] Does this issue include "computing" values for these where missing e.g. for title  and description

## Ideas

* [ ] title
* [ ] date: publication date. would be used in a blog section
* [ ] description: page description
* [ ] image: image to use for the page for example in social preview, in blog post listings etc. TODO: what about setting image dimensions (?)
* [ ] created: date *this* page in wiki was created (so more internal housekeeping)
* [ ] aliases: obsidian field for aliasing pages

Optional ... (to be decided if we use these - if we have them all the time we could have a standard layout for these pages)

* [ ] youtube
* [ ] podcast

**Comment by rufuspollock** [comment]
@PhilippeduPreez any thoughts - also could you share this question on discord.

**Comment by StefanoCecere** [comment]
Obsidian is currently working on improved metadata.. i guess they'll propose some standard fields for date/update/created... what do you think?

**Reply by rufuspollock** [reply]
That would be great if they did. BTW do you have a link to any obsidian forum discussion on this by any chance?

**Comment by StefanoCecere** [comment]
this is how <https://github.com/timlrx/tailwind-nextjs-starter-blog> manages it (copying from Hugo):

```
title (required)
date (required)
tags (required, can be empty array)
lastmod (optional)
draft (optional)
summary (optional)
images (optional, if none provided defaults to socialBanner in siteMetadata config)
authors (optional list which should correspond to the file names in `data/authors`. Uses `default` if none is specified)
layout (optional list which should correspond to the file names in `data/layouts`)
canonicalUrl (optional, canonical url for the post for SEO)
```

with this example:

```
title: 'Introducing Tailwind Nexjs Starter Blog'
date: '2021-01-12'
lastmod: '2021-01-18'
tags: ['next-js', 'tailwind', 'guide']
draft: false
summary: 'Looking for a performant, out of the box template, with all the best in web technology to support your blogging needs? Checkout the Tailwind Nextjs Starter Blog template.'
images: ['/static/images/canada/mountains.jpg', '/static/images/canada/toronto.jpg']
authors: ['default', 'sparrowhawk']
layout: PostLayout
canonicalUrl: https://tailwind-nextjs-starter-blog.vercel.app/blog/introducing-tailwind-nextjs-starter-blog
```
